### DISCUSSION TITLE
Is it possible to rename generated column names?

### DISCUSSION THREAD

**Discussion by rhodeon** [go-jet/jet#278] [Q&A] [answered]
I've been attempting to use Jet with a database which has a table having a column called `alias`.
This becomes a problem when the table file is generated as the column name overrides the `postgres.Table` `Alias()` method.

A simplified example is shown below:
```
type userTable struct {
	postgres.Table

	// Columns
	Alias postgres.ColumnString // this conflicts with the Alias() method of the postgres.Table interface
	Name  postgres.ColumnString

	AllColumns     postgres.ColumnList
	MutableColumns postgres.ColumnList
}
```

Thus, the `FromSchema`  method of the `userTable`:
```
func (a UserTable) FromSchema(schemaName string) *UserTable {
	return newUserTable(schemaName, a.TableName(), a.Alias())
}
```
 results in an error:
`Cannot use 'a.Alias()' (type postgres.ColumnString) as the type string`

Renaming the column in the database itself isn't an option as the database has been mature and in use for many years.

So I wanted to know if it's possible to instead generate the `userTable` with an alternate name for the `alias` column.

From my research, I've seen it's possible to change the types of columns, but I have found nothing about renaming columns.

**Comment by go-jet** [comment, marked as answer]
Hi @rhodeon, it is possible using [generator customization](https://github.com/go-jet/jet/wiki/Generator#generator-customization). Check this [test](https://github.com/go-jet/jet/blob/6a13530ec10b553a6517fb4d1b1c3aae46282dc8/tests/postgres/generator_template_test.go#L536) as example.

**Reply by rhodeon** [reply]
Thanks, that was exactly what I needed.
